core/slice.go

```diff
@@ -505,61 +541,63 @@ func (sl *Slice) computePendingHeader(localPendingHeaderWithTermini types.Pendin
 // updatePhCacheFromDom combines the recieved pending header with the pending header stored locally at a given terminus for specified context
 func (sl *Slice) updatePhCacheFromDom(pendingHeader types.PendingHeader, terminiIndex int, indices []int) error {
 	hash := pendingHeader.Termini[terminiIndex]
-	localPendingHeader, exists := sl.phCache[hash]
+	localPendingHeader, exists := sl.readPhCache(hash)
 
 	if exists {
 		combinedPendingHeader := types.CopyHeader(localPendingHeader.Header)
 		for _, i := range indices {
 			combinedPendingHeader = sl.combinePendingHeader(pendingHeader.Header, combinedPendingHeader, i, false)
 		}
-		sl.writeToPhCacheAndPickPhHead(types.PendingHeader{Header: combinedPendingHeader, Termini: localPendingHeader.Termini}, nil)
-
+		bestPh, exist := sl.readPhCache(sl.bestPhKey)
+		if !exist {
+			log.Error("BestPh Key does not exist for", "key", sl.bestPhKey)
```

Comment:
do we need to handle this, because if doesnt exist it will crash the node

Reply:
will add an else